File: android - Salin/app/src/main/java/com/example/pr_penjualanhandphone_ardisaputratip212/Barang.java
package com.example.pr_penjualanhandphone_ardisaputratip212;

public class Barang extends koneksi{
    server server = new server();
    String conn = server.conn();
    String URL = conn + "/PR_PenjualanHandPhone_ArdiSaputraTIP-21.2/server/barang.php";
    String url = "";
    String response = "";

    public void index(ResponseCallback callback){
        try{
            url = URL + "?operasi=tampil";
            runAsync(url, callback);
        } catch (Exception e){
            e.printStackTrace();
            callback.onFailure(e);
        }
    }
    public String store(String kdbarang, String merk, String tipe, String warna, String ram, String storage, String hargabeli, String hargajual, String stok){
        try {
            url = URL + "?operasi=tambah&kdbarang=" + kdbarang + "&merk=" + merk + "&tipe="  + tipe +
                    "&warna=" + warna + "&ram=" + ram + "&storage=" + storage + "&hargabeli=" + hargabeli +
                    "&hargajual=" + hargajual + "&stok=" + stok;
            response = run(url);
        }catch (Exception e){
            e.printStackTrace();
            e.getMessage();
        }
        return response;
    }
    public String getData(int id){
        try{
            url = URL + "?operasi=getdata&id=" + id;
            response = run(url);
        }catch (Exception e){
            e.printStackTrace();
            e.getMessage();
        }
        return response;
    }
    public String update(String id, String kdbarang, String merk, String tipe, String warna, String ram, String storage, String hargabeli, String hargajual, String stok){
        try{
            url = URL + "?operasi=update&kdbarang=" + kdbarang + "&merk=" + merk + "&tipe="  + tipe +
                    "&warna=" + warna + "&ram=" + ram + "&storage=" + storage + "&hargabeli=" + hargabeli +
                    "&hargajual=" + hargajual + "&stok=" + stok + "&id=" + id;
            response = run(url);
        } catch (Exception e){
            e.printStackTrace();
            e.getMessage();
        }
        return response;
    }
    public String destroy(int kdbarang){
        try {
            url = URL + "?operasi=delete&id=" + kdbarang;
            response = run(url);
        } catch (Exception e){
            e.printStackTrace();
            e.getMessage();
        }
        return response;
    }
    public void search(String key, ResponseCallback callback){
        try{
            url = URL + "?operasi=search&key=" + key;
            runAsync(url, callback);
        } catch (Exception e){
            e.printStackTrace();
            callback.onFailure(e);
        }
    }
    public String selectDataonkd(String key){
        try{
            url = URL + "?operasi=getdatafortransaksi&keykode=" + key;
            response = run(url);
        } catch (Exception e){
            e.printStackTrace();
        }
        return response;
    }
}

File: server/barang.php
<<<<<<< HEAD
<?php
require_once 'DatabaseConnection.php';

$database = new Database();

$tablename = 'barang';
$operasi = $_GET['operasi'];

if (isset($_GET['kdbarang'], $_GET['merk'], $_GET['tipe'], $_GET['warna'], $_GET['ram'], $_GET['storage'], $_GET['hargabeli'], $_GET['hargajual'], $_GET['stok'])) {
    $data = [
        'KDBARANG' => $_GET['kdbarang'],
        'MERK' => $_GET['merk'],
        'TIPE' => $_GET['tipe'],
        'WARNA' => $_GET['warna'],
        'RAM' => $_GET['ram'],
        'STORAGE' => $_GET['storage'],
        'HARGA_BELI' => $_GET['hargabeli'],
        'HARGA_JUAL' => $_GET['hargajual'],
        'STOK_BARANG' => $_GET['stok']
    ];
}

if(isset($_GET['id'])){
    $condition = [
        'IDBARANG' => $_GET['id']
    ];
}


if(isset($_GET['key'])){
    $conditionSearch = [
        'MERK' => $_GET['key'],
    ];
}
if(isset($_GET['keykode'])){
    $conditionTransaksi = [
        'KDBARANG' => $_GET['keykode'],
    ];
}

switch ($operasi) {
    case 'tampil':
        $database->index($tablename);
    break;

    case 'tambah':
        $database->store($data, $tablename);
    break;

    case 'getdata':
        $database->detailData($condition, $tablename);
    break;

    case 'update':
        $database->update($data, $tablename, $condition);
    break;

    case 'delete':
        $database->destroy($condition, $tablename);
    break;

    case 'search':
        // var_dump($conditionSearch);
        $database->searchData($conditionSearch, $tablename);
    break;
    case 'getdatafortransaksi':
        $database->detailData($conditionTransaksi, $tablename);
    break;
    
    default: break;
